### PR TITLE
WD-3695: Data docs

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Continuation of https://github.com/canonical/ubuntu.com/pull/12806
- [Create a single data documentation section](https://github.com/canonical/ubuntu.com/commit/a15c5cbdd58b0cb3caaa3ab9af899b7bbe6ed1f3)

## QA

- Go to [/data/docs](https://ubuntu-com-12969.demos.haus/data/docs) see the docs

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-3695?atlOrigin=eyJpIjoiYTM1NGNiMzQ0ZmM5NDRkMDhlNjFlNDRjMzA2ZTYzYjUiLCJwIjoiaiJ9

### CODE REVIEW

**Comment by anthonydillon** [comment]
@juanruitina would you mind taking a look at this one please when you get a sec?

**Comment by juanruitina** [comment]
Yes! Might not be the perfect solution but it does the job.

Can we keep the URL structure that @mtruj013 and I agreed on [here](https://github.com/canonical/ubuntu.com/pull/12806#issuecomment-1545344857)? It would be like this instead:

/data/docs/kafka/iaas
/data/docs/kafka/k8s

I tweaked the copy doc to incorporate the links to all different documentations, please use the styling we used [here](https://ubuntu.com/kubernetes/docs). Also, please set the first "Home" tab to active.

**Comment by anthonydillon** [comment]
I believe I can replicate the URL structure but we would need another index page for /data/docs/kafka if we do. Another level Ng term option might be to use tabbed content to switch between VM and k8s or versions switching.

**Comment by juanruitina** [comment]
That would be ideal, @grobbie could you provide index pages for each database? They could explain in a paragraph what each database is about, and then link to either option and explain succinctly the differences.

In the meantime, I wouldn't be opposed to just redirecting /data/docs/kafka etc. to /data/docs.

**Comment by grobbie** [comment]
do you want those as pages on discourse or as copydocs to be integrated into u.c?

**Comment by juanruitina** [comment]
> do you want those as pages on discourse or as copydocs to be integrated into u.c?

@anthonydillon What would work better for you?

**Comment by anthonydillon** [comment]
They would need to be pages within the application as that navigate to the specific platform versions of the documentation.

**Comment by anthonydillon** [comment]
@juanruitina Do you think we could progressively get to that state? So to merge this and when we have the new index pages add them and change the URLs?

**Comment by juanruitina** [comment]
I would think so, yes, as long as URLs are the agreed ones. Just link from the top page to the specific documentation in the meantime, as per the [copy doc](https://docs.google.com/document/d/1zidLwSBOsNPIznRO2DfFGU1N3xxnCBKRmQku-X-fc_U/edit).

**Comment by anthonydillon** [comment]
@juanruitina @grobbie I have added all the docs sets to the demo. Please check and let me know if something is missing.

**Comment by juanruitina** [comment]
This is it! ✌️

For some reason the navigation [for Charmed MySQL](https://ubuntu-com-12969.demos.haus/data/docs//mysql/iaas) isn't showing up.

And some minor changes:

- [ ] All links have a double backlash after `docs`, e.g. `/data/docs//mysql/iaas`. Can we remove the extra one?
- [ ] On the left sidebar, please add the link to the Spark one inside the toggle: [Spark on Kubernetes](https://ubuntu.com/data/docs/spark/k8s)
- [ ] On the text body on the right, make the database name/title bold (e.g. "Charmed MySQL") as per the copy doc.
- [ ] On the right, add the link to the Spark one below the description, as a single list item: [Spark on Kubernetes](https://ubuntu.com/data/docs/spark/k8s) (I just updated the copy doc)
- [ ] There's a rogue `p-strip` above the "Last updated" notification, can you remove it?

@grobbie There are some issues within the different documentations, I understand these can be addressed on Discourse:

* Some repeated headings, [e.g.](https://ubuntu-com-12969.demos.haus/data/docs//kafka/iaas). It seems the docs take the title of the Discourse topic, so no need to add it to the text body.
* Please drop the overview headings within the text body [here](https://ubuntu-com-12969.demos.haus/data/docs//mongodb/k8s) and [here](https://ubuntu-com-12969.demos.haus/data/docs//mongodb/iaas).
* The Overview pages (the main ones) are missing from the navigation, on all except [Spark](https://ubuntu-com-12969.demos.haus/data/docs//spark/k8s). Can you make sure to add them for all others?

**Comment by grobbie** [comment]
@deusebio can you address Juan's remarks on the discourse posts please? Thanks

**Comment by grobbie** [comment]
Great start. One remark but the docs search function doesn't appear to work?

**Comment by juanruitina** [comment]
@anthonydillon Taking into account @deusebio's comments, do you think we can make the headings look good both on ubuntu.com and Charmhub, and add the "Overview" link to the side navigation?

**Comment by anthonydillon** [comment]
Sure, I have disabled the title and only rendered the content for these docs.

File: templates/data/spark/index.html
Comment (mtruj013) [inline templates/data/spark/index.html:66]:
```suggestion
            loading="lazy"
```

File: templates/data/spark/index.html
Comment (mtruj013) [inline templates/data/spark/index.html:81]: Can be deleted

File: templates/data/spark/index.html
Comment (mtruj013) [inline templates/data/spark/index.html:125]:
```suggestion
        loading="lazy"
```

File: templates/shared/forms/interactive/data-docs.html
Comment (mtruj013) [inline templates/shared/forms/interactive/data-docs.html:1]: I can't find where this form is used, where can I QA?

File: templates/data/docs/kafka/iaas/document.html
Comment (mtruj013) [inline templates/data/docs/kafka/iaas/document.html:1]: Is this the expected behavior for this one? 
Screenshot from 2023-07-03 10-38-02


There's no sidenav and the step by step guide are direct links to discourse, in contrast to the rest of the the tutorials which look like so:
image

File: templates/data/docs/opensearch/iaas/document.html
Comment (mtruj013) [inline templates/data/docs/opensearch/iaas/document.html:1]: Double heading on this one 
Screenshot from 2023-07-03 10-40-46

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks @anthonydillon! Some comments in the code:

File: templates/shared/forms/interactive/data-docs.html
Comment (anthonydillon) [reply templates/shared/forms/interactive/data-docs.html:1]: I don't see it either. WIll remove for now.

File: templates/data/docs/kafka/iaas/document.html
Comment (anthonydillon) [reply templates/data/docs/kafka/iaas/document.html:1]: @deusebio can we add a navigation to this tutorial?

**Comment by juanruitina** [comment]
@anthonydillon Can you please incorporate the changes I mentioned [above](https://github.com/canonical/ubuntu.com/pull/12969#issuecomment-1608922856) so I can do a full review?

**Comment by anthonydillon** [comment]
@juanruitina @mtruj013 all changes applied

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks @anthonydillon. Few more things: 

- The titles are now missing for `/mysql/k8s`, `/postgresql/iaas`, and `/postgresql/k8s`. 
- I didn't catch this before, but is `Charmed Spark` not supposed to have a dropdown like the rest? At first I thought it was maybe because it only includes a single page, but that's not the case
image
image
- The double `\\` in the URL is still showing up for me

File: templates/data/docs/kafka/iaas/document.html
Comment (deusebio) [reply templates/data/docs/kafka/iaas/document.html:1]: Yes. Definitely. Let me follow up on this and give you the heads up once it is fixed

File: webapp/app.py
Comment (taurus-forever) [inline webapp/app.py:682]: The correct id is [9925](https://discourse.charmhub.io/t/charmed-mysql-documentation/9925).

File: templates/data/docs/kafka/iaas/document.html
Comment (taurus-forever) [reply templates/data/docs/kafka/iaas/document.html:1]: The mentioned link: https://ubuntu-com-12969.demos.haus/data/docs//mysql/iaas points to MySQL "Tutorial"
https://discourse.charmhub.io/t/charmed-mysql-tutorial-overview/9922

The correct high-level link is https://discourse.charmhub.io/t/charmed-mysql-documentation/9925 
The correct link used in the charm: https://github.com/canonical/mysql-operator/blob/main/metadata.yaml#L13C7-L13C71

See https://github.com/canonical/ubuntu.com/pull/12969#pullrequestreview-1511173616

**Comment by taurus-forever** [comment]
> * The titles are now missing for `/mysql/k8s`, `/postgresql/iaas`, and `/postgresql/k8s`. (Update: `/mysql/iaas` is the same)

CC: @deusebio I removed them in purpose, to have nice charm landing page:
Screenshot from 2023-07-03 15-16-20

I can restore titles on Discourse, but can they be hidden on charmhub? Otherwise no beauty on the landing page:
Screenshot from 2023-07-03 15-19-10


P.S. As I can see the title is also missing for [MongoDB VM/IaaS](https://ubuntu-com-12969.demos.haus/data/docs//mongodb/iaas).

**Comment by anthonydillon** [comment]
I've had a look and I believe the double slash in the URL is a bug in the discourse documentation module when switching between docs sets.

File: webapp/app.py
Comment (anthonydillon) [reply webapp/app.py]: Thanks, that is better :+1:

**Comment by anthonydillon** [comment]
@deusebio @juanruitina do you want headings on the docs templates restored? I removed them as they were duplicating but seems the source have changed.

**Comment by juanruitina** [comment]
UX-wise, I would need every page to have exactly one H1 heading as the title, but I defer to @deusebio and you about the best technical way to achieve that.

I see some of the [feedback I provided](https://github.com/canonical/ubuntu.com/pull/12969#issuecomment-1608922856) is still not implemented. Let me know if any issue arises when incorporating it.

**Comment by anthonydillon** [comment]
@taurus-forever @deusebio did you want the title of the topic rendered as a top heading or leave it just as the content of the topic?

**Comment by deusebio** [comment]
Hi @anthonydillon ! I have just synced with @taurus-forever this morning. In order to keep things consistent with Charmhub (where the title is not rendered) it will be best to not have the title of the topic rendered. So, @taurus-forever will just re-introduce the titles across in Discourse

**Comment by anthonydillon** [comment]
Great thanks, I will double check all docs have no titles.

**Comment by deusebio** [comment]
> * [x] On the left sidebar, please add the link to the Spark one inside the toggle: [Spark on Kubernetes](https://ubuntu.com/data/docs/spark/k8s)

Fixed!

> * [x]  On the text body on the right, make the database name/title bold (e.g. "Charmed MySQL") as per the [copy doc](https://docs.google.com/document/d/1zidLwSBOsNPIznRO2DfFGU1N3xxnCBKRmQku-X-fc_U/edit).

after looking at the docs, I realized that the sections, e.g. CharmedMySQL, has links in it, pointing to tha product main pages, e.g. https://ubuntu.com/data/docs/spark/k8s, that will exists only once that the webpage will be live. I have updated the links, but these are currently broken. Let me know if I have to revert this

> * [x]  On the right, add the link to the Spark one below the description, as a single list item: [Spark on Kubernetes](https://ubuntu.com/data/docs/spark/k8s) (I just updated the copy doc)

Fixed!

> So, @taurus-forever will just re-introduce the titles in the content across the products in Discourse

@taurus-forever just give us the heads up once you have addressed these, such that a final check can be done

**Comment by juanruitina** [comment]
> after looking at the docs, I realized that the sections, e.g. CharmedMySQL, have links in them, pointing to the product main pages, e.g. https://ubuntu.com/data/docs/spark/k8s, that will exists only once that the webpage will be live. I have updated the links, but these are currently broken. Let me know if I have to revert this

Thank you @deusebio. Given that it doesn't seem we will have the pages yet, please remove the links from those titles and just make them bold. I tweaked the [copy doc](https://docs.google.com/document/d/1zidLwSBOsNPIznRO2DfFGU1N3xxnCBKRmQku-X-fc_U/edit) to avoid any confusion; we can readd those links whenever we have the content ready.

**Comment by deusebio** [comment]
> Thank you @deusebio. Given that it doesn't seem we will have the pages yet, please remove the links from those titles and just make them bold. I tweaked the [copy doc](https://docs.google.com/document/d/1zidLwSBOsNPIznRO2DfFGU1N3xxnCBKRmQku-X-fc_U/edit) to avoid any confusion; we can re-add those links whenever we have the content ready.

@juanruitina Done! ;)

**Comment by juanruitina** [comment]
Fantastic, /data/docs looks good now.

> Hi @anthonydillon ! I have just synced with @taurus-forever this morning. In order to keep things consistent with Charmhub (where the title is not rendered) it will be best to not have the title of the topic rendered. So, @taurus-forever will just re-introduce the titles in the content across the products in Discourse

@deusebio @taurus-forever Please let me know when the title issues are addressed on Discourse so I can +1 this all. Let's aim for some consistency, mentioning the product in the H1, [this one is a good example](https://ubuntu-com-12969.demos.haus/data/docs//opensearch/iaas). "Overview" headings might not be necessary as they are a bit redundant, mentioning the product name is more meaningful.

**Review by mtruj013 — APPROVED** [review]
LGTM! The path on `/mysql/iaas` > Reference > Libraries needs to be fixed as it's currently a 404, but that's a discourse update so I think all good from our side 👍🏻

**Comment by anthonydillon** [comment]
@deusebio would you be able to address the content issues above ^

**Comment by taurus-forever** [comment]
I have synced the title for all components as "Charmed Component (K8s)? Documentation".
Please check it from your side.

P.S. I dislike title on charmhub, see https://charmhub.io/mysql (the message "Charmed MySQL Documentation" is huge and alien there). Maybe someone can find 10 mins to polish CSS there, see: https://github.com/canonical/charmhub.io/issues/1574 
Tnx!

**Comment by juanruitina** [comment]
@taurus-forever Thank you for the changes. Can you please change the title under Kafka and Kafka K8s documentation so they are H1s instead of H2s? Other than that, this looks good to me now.

I see what you mean about the sizing of headings, I'll follow up with colleagues working on Charmhub.

**Comment by juanruitina** [comment]
I see the issue with the Kafka docs has been addressed, so this LGTM now.

**Pull request merged**
